### PR TITLE
[COLLECTIONS-809] JUnit v5 assertThrows

### PR DESCRIPTION
**Pull request by nhojpatrick**
Refactor PredicatedMapTest

Refactor UnmodifiableSortedSetTest

Refactor PredicatedSortedSetTest

Refactor PredicatedSetTest

Refactor PredicatedNavigableSetTest

Refactor ListOrderedSetTest

Refactor PredicatedSortedMapTest

Refactor ReferenceIdentityMapTest

Refactor ReferenceMapTest

Refactor TransformedSortedMapTest

### CODE REVIEW

File: src/test/java/org/apache/commons/collections4/map/TransformedSortedMapTest.java [hunk 1]
```diff
@@ -77,18 +79,15 @@ public void testTransformedMap() {
             map.put((K) els[i], (V) els[i]);
             assertEquals(i + 1, map.size());
             assertTrue(map.containsKey(Integer.valueOf((String) els[i])));
-            try {
-                map.containsKey(els[i]);
-                fail();
-            } catch (final ClassCastException ex) {}
+            SortedMap<K, V> finalMap1 = map;
+            int finalI = i;
```
Comment (kinow) [inline src/test/java/org/apache/commons/collections4/map/TransformedSortedMapTest.java:83]: Why did you need these two variables above? :thinking:

**Review by kinow — COMMENTED** [review]
One quick question, but the rest looks great. Thanks!

File: src/test/java/org/apache/commons/collections4/map/TransformedSortedMapTest.java (hunk 1 above)
Comment (nhojpatrick) [reply src/test/java/org/apache/commons/collections4/map/TransformedSortedMapTest.java:83]: Otherwise you get this error "Variable used in lambda expression should be final or effectively final"

**Comment by nhojpatrick** [comment]
Moving to draft and will keep merging all the others into it.
